auth_service/auth_adapter/api/basic.py

```diff
@@ -28,16 +29,52 @@
 __all__ = ["basic_auth"]
 
 
+def get_allowed_credentials(config: Config) -> list[HTTPBasicCredentials]:
+    """Get list of allowed credentials from the config.
+
+    Multiple users and passwords can be specified separated with commas.
+    Passwords can be specified in the user setting separated with colons.
+    """
+    user, password = config.basic_auth_user, config.basic_auth_pwd
+    try:
+        if user:
+            users = user.split(",")
+            if password:
+                passwords = password.split(",")
+                if len(passwords) != len(users):
+                    raise ValueError(
+                        "Different numbers of Basic Auth users and passwords."
+                    )
+                user_passwords = cast(list[list[str]], zip(users, passwords))
+            else:
+                user_passwords = [user.split(":", 1) for user in users]
+            allowed_credentials = [
+                HTTPBasicCredentials(username=user.strip(), password=password.strip())
+                for (user, password) in user_passwords
+            ]
+            if not all(
+                credentials.username and credentials.password
+                for credentials in allowed_credentials
+            ):
+                raise ValueError("Empty Basic Auth user or password.")
+        elif password:
+            raise ValueError("Basic Auth password with no user.")
+        else:
+            allowed_credentials = []
```

Comment:
This validation might better be in the config model, as demonstrated here: #11

Reply:
That's true. The suggestion to use secret strings is also a good one, and I changed that already.

For now, I'll leave the settings "hack" to add multiple credentials. It's also a bit easer to configure and backward compatible.